### PR TITLE
feat: new set of parsers

### PR DESCRIPTION
**Pull request by Seddryck**
## Summary by CodeRabbit

- **New Features**
  - Introduced a modular CSV parsing framework with new parser interfaces and implementations supporting advanced dialect features including arrays, comments, quoting, escaping, and custom delimiters.
  - Added support for specifying missing cell representation in CSV dialect configuration.
  - Enhanced record state tracking with explicit states for comments, headers, and end-of-file.
- **Bug Fixes**
  - Improved null safety and consistent field span handling in parsing and data access.
- **Refactor**
  - Replaced legacy character parser with a context-controller based parser architecture.
  - Redesigned field span structures to encapsulate metadata more effectively.
  - Refactored tests and internal logic to align with the new parsing system and updated APIs.
- **Tests**
  - Added extensive unit tests covering new parser components, dialect variations, and edge cases such as arrays, comments, quoting, and escaping.
- **Chores**
  - Removed obsolete code and project references related to deprecated modules.
  - Updated CI configuration to reflect project and solution structure changes.
- **Documentation**
  - Added XML documentation comments to public APIs and improved error messages for clarity on row and field handling.

### CODE REVIEW

File: PocketCsvReader/CharParsing/CommentParser.cs [hunk 1]
```diff
@@ -1,23 +1,44 @@
...
+    private readonly IParserStateController _controller;
 
-    public override ParserState NextState()
-        => ParserState.Continue;
+    private readonly char _lineTerminatorChar;
+    private readonly int _lineTerminatorLength;
 
-    public override ParserState SetBack()
+    public CommentParser(IParserContext ctx, IParserStateController controller, string lineTerminator)
+        => (_ctx, _controller, _lineTerminatorLength, _lineTerminatorChar) = (ctx, controller, lineTerminator.Length, lineTerminator[0]);
```
[the remark this answers is not in the document]
Comment (Seddryck) [reply PocketCsvReader/CharParsing/CommentParser.cs:17]: I understand but the handling of the full multi-char terminator is delegated to the lineTerminatorParser. No?

File: PocketCsvReader/CharParsing/LineTerminatorParser.cs [hunk 2]
```diff
@@ -1,53 +1,52 @@
...
+        }
+        else
+        {
+            Reset();
+            _controller.SwitchBack();
             return ParserState.Continue;
         }
-
-        return SetBack();
     }
```
[the remark this answers is not in the document]
Comment (Seddryck) [reply PocketCsvReader/CharParsing/LineTerminatorParser.cs:35]: Yes, the first char is checked in commentParser or rawParser ...

File: PocketCsvReader/BaseRecordParser.cs [hunk 3]
```diff
@@ -33,36 +35,55 @@ public virtual bool IsEndOfFile(out RecordSpan record)
...
 
                 if (state == ParserState.Record || state == ParserState.Header)
                 {
-                    CharParser.Reset();
                     Buffer = Buffer.Slice(index + 1);
                     FieldsCount ??= fieldList.Count;
                     record = CreateRecordSpan(
                         longSpan.Length > 0 ? (ReadOnlySpan<char>)(longSpan.Concat(span)) : span
                         , [.. fieldList]);
+                    recordState = RecordState.Record;
```
[the remark this answers is not in the document]
Comment (Seddryck) [reply PocketCsvReader/BaseRecordParser.cs:64]: I know that headers are not well handled
